flocker/provision/_aws.py

```diff
@@ -27,70 +39,227 @@
 }
 
 
-def get_default_username(distribution):
+IMAGE_NAMES = {
+    'centos-7': 'CentOS 7 x86_64 (2014_09_29) EBS HVM'
+                '-b7ee8a69-ee97-4a49-9e68-afaee216db2e-ami-d2a117ba.2',
+    'ubuntu-14.04': 'ubuntu/images/hvm-ssd/ubuntu-trusty-14.04-amd64-server-20150325',  # noqa
+    'ubuntu-15.04': 'ubuntu/images/hvm-ssd/ubuntu-vivid-15.04-amd64-server-20150422',  # noqa
+}
+
+
+def _wait_until_running(instance):
     """
-    Return the username available by default on a system.
+    Wait until a instance is running.
 
-    :param str distribution: Name of the operating system distribution
-    :return str: The username made available by AWS for this distribution.
+    :param boto.ec2.instance.Instance instance: The instance to wait for.
     """
-    return _usernames[distribution]
+    with start_action(
+        action_type=u"flocker:provision:aws:wait_until_running",
+        instance_id=instance.id,
+    ):
+        state = instance.state
+        while state != 'running':
+            with start_action(
+                action_type=u"flocker:provision:aws:wait_until_running:sleep",
+                instance_state=instance.state,
+            ):
+                sleep(1)
+            instance.update()
+            new_state = instance.state
```

Comment:
:question: `state = instance.state`? `new_state` seems unnecessary (left over from earlier version of code maybe.)